pypesto/hierarchical/spline_approximation/calculator.py

```diff
@@ -0,0 +1,215 @@
+import copy
+from typing import Dict, List, Sequence, Tuple
+
+import numpy as np
+
+from ...C import FVAL, GRAD, HESS, MODE_RES, RDATAS, RES, SRES, X_INNER_OPT
+from ...objective.amici.amici_calculator import AmiciModel, AmiciSolver
+from ...objective.amici.amici_util import (
+    filter_return_dict,
+    init_return_values,
+)
+from .problem import SplineInnerProblem
+from .solver import SplineInnerSolver
+
+try:
+    import amici
+    from amici.parameter_mapping import ParameterMapping
+except ImportError:
+    pass
+
+
+class SplineAmiciCalculator:
```

Comment:
Should I do this even if the `SplineAmiciCalculator` redefines all functions from `AmiciCalculator` (there are only `initialize` and `__call__`) , and there are no other functions of `AmiciCalculator`? Since there is nothing really it inherits from  `AmiciCalculator`.

Reply:
it's supposed to be used inside the objective as a calculator, right? https://github.com/ICB-DCM/pyPESTO/blob/main/pypesto/objective/amici/amici.py#L66 Then it needs to be derived from the base class.

Reply:
True True, will add for both ordinal and spline